arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/transients/StackHeightCheck.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Consensys Software Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package net.consensys.linea.zktracer.module.hub.transients;
+
+import static com.google.common.base.Preconditions.*;
+import static net.consensys.linea.zktracer.runtime.stack.Stack.MAX_STACK_SIZE;
+
+import lombok.EqualsAndHashCode;
+import net.consensys.linea.zktracer.container.ModuleOperation;
+
+@EqualsAndHashCode(onlyExplicitlyIncluded = true, callSuper = false)
+public class StackHeightCheck extends ModuleOperation {
+  private static final int SHIFT_FACTOR =
+      8; // 5 would suffice but 8 makes it a byte shift, not sure if it matters
+
+  @EqualsAndHashCode.Include final int comparison;
+
+  @Override
+  protected int computeLineCount() {
+    return 0;
```

Comment:
instead of 0 it should return an exception, as this line count shouldn't be called